packages/devextreme/js/ui/scheduler/ui.scheduler.js

```diff
@@ -1352,10 +1353,26 @@ class Scheduler extends Widget {
                 appointment,
                 this._dataAccessors,
                 this.timeZoneCalculator
-            ).disabled
+            ).disabled,
+            onItemContextMenu: that._createActionByOption('onAppointmentContextMenu'),
+            createEventArgs: that._createEventArgs.bind(that),
         };
     }
 
+    _createEventArgs(e) {
+        const config = {
+            itemData: e.itemData.appointment,
+            itemElement: e.itemElement,
+            targetedAppointment: e.itemData.targetedAppointment,
+        };
+        return extend({}, this.fire('mapAppointmentFields', config), {
```

Comment:
Can **extend function** be replaced by the **spread operator**?

Reply:
The [extend function's ability to exclude null fields](https://github.com/DevExpress/DevExtreme/blob/23_2/packages/devextreme/js/core/utils/extend.js#L30) is required here. For instance, [onAppointmentContextMenu](https://js.devexpress.com/Documentation/ApiReference/UI_Components/dxScheduler/Configuration/#onAppointmentContextMenu)'s `model` event property is conditionally available, and so must be defined in the resulting event object only if available.

Technically, the same logic can be done with spread operator (e.g., implement the same if statement in the extend function), but I think it's better to just reuse available utility methods.